### PR TITLE
init basic quarkus server

### PR DESCRIPTION
**Pull request by hjmodi**

### CODE REVIEW

File: src/main/java/org/commonjava/indy/service/httprox/HttpProxy.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+        XnioWorker worker;
+        try {
+            worker = Xnio.getInstance()
+                    .createWorker(OptionMap.EMPTY);
+
+            final InetSocketAddress addr;
+            if (config.getPort() < 1) {
+                ThreadLocal<InetSocketAddress> using = new ThreadLocal<>();
+                ThreadLocal<IOException> errorHolder = new ThreadLocal<>();
```
Comment (ligangty) [inline src/main/java/org/commonjava/indy/service/httprox/HttpProxy.java:74]: Seems this errorHolder is not used?

**Review by ligangty — COMMENTED** [review]
LGTM.

**Comment by ligangty** [comment]
> @ruhan1 @ligangty @jdcasey The original plan was to only start a skeleton service(it still is but this is bigger code) but I figured having the additional classes to create proper responses will be helpful later.
> 
> It's not a 100% ideal configuration for starting proxy server in quarkus. Please let me know if you know of a better way.

@hjmodi so I'm not sure what's the cons. you are thinking that the quarkus is not a good choice here. Can you help share your concerns?

**Comment by ruhan1** [comment]
I guess this is the init part of the httproxy and basing on existing code. It looks good. But I have same question as whether we need Quarkus to be here. As my first look, Quarkus only plays a little bit in the startup/shutdown event. I really don't know what other benefit would come from Quarkus ecosystem. And in this case, we don't really need to compile to native.

File: src/main/java/org/commonjava/indy/service/httprox/HttpProxy.java (hunk 1 above)
Comment (hjmodi) [reply src/main/java/org/commonjava/indy/service/httprox/HttpProxy.java:74]: remove it.

**Comment by hjmodi** [comment]
> > @ruhan1 @ligangty @jdcasey The original plan was to only start a skeleton service(it still is but this is bigger code) but I figured having the additional classes to create proper responses will be helpful later.
> > It's not a 100% ideal configuration for starting proxy server in quarkus. Please let me know if you know of a better way.
> 
> @hjmodi so I'm not sure what's the cons. you are thinking that the quarkus is not a good choice here. Can you help share your concerns?

I am just wondering if this is the best way to do it since we're not utilizing quarkus much but I'm thinking of continuing with quarkus and remove it later if it's still underutilized.

**Pull request merged**